DataFormats/interface/Photon.h

```diff
@@ -85,7 +88,7 @@ namespace flashgg {
 
     bool hasEnergyAtStep(std::string key) const;
     float const energyAtStep(std::string key) const;
-    float const sigEOverE() const { return sigEOverE_; };
+    float const sigEOverE() const { return (getCorrectedEnergyError(regression_type) / energy()); } // Regression + error now in reco and base object
```

Comment:
Isn't there a sigmaEoverE variable already stored in reco?

Computing sigmaEoverE as sigmaE/E has the problem that the denominator changes upon application of energy scale corrections/smearing.

If the relative resolution is not available, I would repalce energy() with the appropriate energyAtStep() call.